google/cloud/spanner_v1/session.py

```diff
@@ -460,6 +462,21 @@ def run_in_transaction(self, func, *args, **kw):
         exclude_txn_from_change_streams = kw.pop(
             "exclude_txn_from_change_streams", None
         )
+        isolation_level = kw.pop("isolation_level", None)
+
+        if isolation_level is None:
+            isolation_level = (
+                self._database.default_transaction_options.get(
+                    "isolation_level",
+                    TransactionOptions.IsolationLevel.ISOLATION_LEVEL_UNSPECIFIED,
+                )
+                if isinstance(self._database.default_transaction_options, dict)
+                else self._database.default_transaction_options.isolation_level
+                if isinstance(
+                    self._database.default_transaction_options, TransactionOptions
+                )
+                else TransactionOptions.IsolationLevel.ISOLATION_LEVEL_UNSPECIFIED
+            )
```

Comment:
this seems to be a repetion of the same code block from above. Could we add a function to `Database` to encapsulate this in one place?

Reply:
Done. Extracted into a method